src/provider/provider_level_zero.c

```diff
@@ -366,8 +366,15 @@ static umf_result_t ze_memory_provider_initialize(void *params,
         return UMF_RESULT_ERROR_INVALID_ARGUMENT;
     }
 
-    if ((ze_params->memory_type == UMF_MEMORY_TYPE_HOST) ==
+    if ((ze_params->memory_type == UMF_MEMORY_TYPE_HOST) &&
         (ze_params->level_zero_device_handle != NULL)) {
+        LOG_ERR("For HOST memory type the Level Zero device handle "
+                "must be NULL");
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    if ((ze_params->memory_type != UMF_MEMORY_TYPE_HOST) &&
+        (ze_params->level_zero_device_handle == NULL)) {
```

Comment:
Does this change should belong to this PR? What does it have in common?

Reply:
reverted